wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/LanguageServerConfig.java

```diff
@@ -0,0 +1,189 @@
+   * launches a language server process correspondingly.
+   *
+   * @see EmptyCommunicationProvider
+   * @see SocketCommunicationProvider
+   * @see ProcessCommunicationProvider
+   */
+  interface CommunicationProvider {
+    /**
+     * Get input stream
+     *
+     * @return input stream
+     * @throws LanguageServerException if any issue happened trying to get an input stream
+     */
+    InputStream getInputStream() throws LanguageServerException;
+
+    /**
+     * Get output stream
+     *
+     * @return output stream
+     * @throws LanguageServerException if any issue happened trying to get an output stream
+     */
+    OutputStream getOutputStream() throws LanguageServerException;
+
+    /**
+     * Get status checker
+     *
+     * @return status checker
+     * @throws LanguageServerException if any issue happened trying to get a status checker
+     */
+    StatusChecker getStatusChecker() throws LanguageServerException;
+
+    /**
+     * The implementation must provide aliveness status for language server that we communicate to.
+     */
+    interface StatusChecker {
+      /**
+       * Shows if a language server communication is alive
+       *
+       * @return true - if communication is alive, false - if communication is not alive
+       */
+      boolean isAlive();
+
+      /**
+       * Get cause of language server communication not being alive.
+       *
+       * @return short description of why the communication is not alive, if the communication is
+       *     alive (<code>isAlive == true</code>) returns <code>null</code>.
+       */
+      String getCause();
+    }
+  }
+
+  /** Provides language server matching regular expression definitions. */
+  interface RegexProvider {
+    /**
+     * Get a map of <code>languageId->languageRegexp</code> where <code>languageId</code> is the
+     * identifier of a language that is listed in the <a
+     * href=https://microsoft.github.io/language-server-protocol/specification>language server
+     * specification</a>, while <code>languageRegexp</code> is a regular expression that matches all
+     * files that are related to a corresponding language. This map is used to determine which file
+     * belongs to which language server.
+     *
+     * @return map of language regexes
+     */
+    Map<String, String> getLanguageRegexes();
+
+    /**
+     * Get a file watch patterns for a {@link LanguageServerFileWatcher}
+     *
+     * @return set of stringified file watch patterns
+     */
+    Set<String> getFileWatchPatterns();
+  }
+
+  /**
+   * Some language servers may be configured to be launched locally with the help of installers and
+   * process launchers. In order to be aware of installer's actions one must implement an installer
+   * status provider that will show if installer performed all required installations i.e. language
+   * server and all related dependencies has been properly installed.
+   */
+  interface InstallerStatusProvider {
```

Comment:
Yeah, I agree. Moreover in fact there is no good way or mechanism how we can check/get status of an installer and its progression. Do you have any specific idea how it can be improved?